src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -52,79 +61,65 @@
  */
 public class EntityCitizen extends EntityAgeable implements INpc
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 60 to the maximum authorized 20 or less. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') class "EntityCitizen" has 92 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)